webapp/store/views.py

```diff
@@ -518,3 +518,24 @@ def entity_icon(entity_name):
         "https://res.cloudinary.com/canonical/image/fetch/f_auto"
         f",q_auto,fl_sanitize,w_64,h_64/{icon_url}"
     )
+
+
+# This method is a temporary hack to show bundle icons on the
+# homepage, and should be removed once the icons are available via the api
+@store.route(
+    '/<regex("' + DETAILS_VIEW_REGEX + '"):entity_name>/bundle-apps.json'
```

Comment:
```suggestion
    '/<regex("' + DETAILS_VIEW_REGEX + '"):entity_name>/charms.json'
```

This should work if the entity is a bundle.